components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -138,4 +169,18 @@ public void reindex()
             this.logger.warn("Failed to search patients for reindexing: {}", ex.getMessage());
         }
     }
+
+    private void addGenesWithStatusToField(SolrInputDocument input, PatientData<Map<String, String>> allGenes,
+        String status, String field)
+    {
+        if (allGenes != null && allGenes.isIndexed()) {
```

Comment:
isIndexed() is probably unnecessary as we already require PatientData to be of the Map<String,String> kind (Sergiu should kno better if this is a good idea to keep it?)